Detectors/Base/include/DetectorsBase/Vector.h

```diff
@@ -0,0 +1,395 @@
+#ifndef O2_BASE_VECTOR_H
+#define O2_BASE_VECTOR_H
+
+#include <TTree.h>
+#include <TBranch.h>
+#include <TBranchElement.h>
+#include <TBuffer.h>
+#include <TObject.h>
+#include <memory>
+#include <algorithm>
+#include <type_traits>
+#include <stdexcept>
+#include <iostream>
+#include <FairLogger.h>
+
+
+namespace o2 {
+namespace Base {
+
+ template <class T, class H>
+   class Vector: public TObject {
```

Comment:
@sawenzel : for the TObject: indeed, thanks for the suggestion, fix pushed. For the name: originally it was vectCont, was changed to Vector acc. reviewer suggestion :) Let's at the moment discuss more global problems, the cosmetics can be done later.